groupie-core/src/main/java/com/xwray/groupie/BaseGroupAdapter.java

```diff
@@ -0,0 +1,269 @@
+package com.xwray.groupie;
+
+import android.support.annotation.NonNull;
+import android.support.v7.widget.GridLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.ViewGroup;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * An adapter that holds a list of Groups.
+ */
+public abstract class BaseGroupAdapter<I extends BaseItem, VH extends BaseViewHolder<I>> extends RecyclerView.Adapter<VH> implements GroupDataObserver {
+
+    private final List<Group<I>> groups = new ArrayList<>();
+    private BaseOnItemClickListener<I> onItemClickListener;
+    private BaseOnItemLongClickListener<I> onItemLongClickListener;
+    private int spanCount = 1;
+    
+    private final GridLayoutManager.SpanSizeLookup spanSizeLookup = new GridLayoutManager.SpanSizeLookup() {
+        @Override public int getSpanSize(int position) {
+            try {
+                return getItem(position).getSpanSize(spanCount, position);
+            } catch (IndexOutOfBoundsException e) {
+                // Bug in support lib?  TODO investigate further
+                return spanCount;
+            }
+        }
+    };
+
+    public GridLayoutManager.SpanSizeLookup getSpanSizeLookup() {
+        return spanSizeLookup;
+    }
+
+    public void setSpanCount(int spanCount) {
+        this.spanCount = spanCount;
+    }
+
+    public int getSpanCount() {
+        return spanCount;
+    }
+
+    /**
+     * Optionally register an {@link BaseOnItemClickListener} that listens to click at the root of
+     * each Item where {@link BaseItem#isClickable()} returns true
+     * @param onItemClickListener The click listener to set
+     */
+    public void setOnItemClickListener(BaseOnItemClickListener<I> onItemClickListener) {
+        this.onItemClickListener = onItemClickListener;
+    }
+
+    /**
+     * Optionally register an {@link BaseOnItemLongClickListener} that listens to long click at the root of
+     * each Item where {@link BaseItem#isLongClickable()} returns true
+     * @param onItemLongClickListener The long click listener to set
+     */
+    public void setOnItemLongClickListener(BaseOnItemLongClickListener<I> onItemLongClickListener) {
+        this.onItemLongClickListener = onItemLongClickListener;
+    }
+
+    @Override public abstract VH onCreateViewHolder(ViewGroup parent, int layoutResId);
+
+    @Override public void onBindViewHolder(VH holder, int position) {
+        // Never called (all binds go through the version with payload)
```

Comment:
Maybe throw an exception in case it is called erroneously to help diagnose problem.